SimplifiedTests/NYPLReaderBookmarksBusinessLogicTests.swift

```diff
@@ -0,0 +1,214 @@
+//
+//  NYPLReaderBookmarkBusinessLogicTests.swift
+//  SimplyETests
+//
+//  Created by Ernest Fan on 2020-10-29.
+//  Copyright © 2020 NYPL Labs. All rights reserved.
+//
+
+import XCTest
+import R2Shared
+@testable import SimplyE
+
+class NYPLReaderBookmarksBusinessLogicTests: XCTestCase {
+    var bookmarkBusinessLogic: NYPLReaderBookmarksBusinessLogic!
+    var bookRegistryMock: NYPLBookRegistryMock!
+    var libraryAccountMock: NYPLLibraryAccountMock!
+    var bookmarkCounter: Int = 0
+    let bookIdentifier = "fakeEpub"
+    
+    override func setUpWithError() throws {
+      try super.setUpWithError()
+      
+      let emptyUrl = URL.init(fileURLWithPath: "")
+      let fakeAcquisition = NYPLOPDSAcquisition.init(
+        relation: .generic,
+        type: "application/epub+zip",
+        hrefURL: emptyUrl,
+        indirectAcquisitions: [NYPLOPDSIndirectAcquisition](),
+        availability: NYPLOPDSAcquisitionAvailabilityUnlimited.init()
+      )
+      let fakeBook = NYPLBook.init(
+        acquisitions: [fakeAcquisition],
+        bookAuthors: [NYPLBookAuthor](),
+        categoryStrings: [String](),
+        distributor: "",
+        identifier: bookIdentifier,
+        imageURL: emptyUrl,
+        imageThumbnailURL: emptyUrl,
+        published: Date.init(),
+        publisher: "",
+        subtitle: "",
+        summary: "",
+        title: "",
+        updated: Date.init(),
+        annotationsURL: emptyUrl,
+        analyticsURL: emptyUrl,
+        alternateURL: emptyUrl,
+        relatedWorksURL: emptyUrl,
+        seriesURL: emptyUrl,
+        revokeURL: emptyUrl,
+        report: emptyUrl
+      )
+      
+      bookRegistryMock = NYPLBookRegistryMock()
+      bookRegistryMock.addBook(book: fakeBook, state: .DownloadSuccessful)
+      libraryAccountMock = NYPLLibraryAccountMock()
+      
+      bookmarkBusinessLogic = NYPLReaderBookmarksBusinessLogic(
+        book: fakeBook,
+        r2Publication: Publication(metadata: Metadata(title: "fakeMetadata")),
+        drmDeviceID: "fakeDeviceID",
+        bookRegistryProvider: bookRegistryMock,
+        currentLibraryAccountProvider: libraryAccountMock)
+      bookmarkCounter = 0
+    }
+
+    override func tearDownWithError() throws {
+      try super.tearDownWithError()
+      bookmarkBusinessLogic = nil
+      libraryAccountMock = nil
+      bookRegistryMock.identifiersToRecords.removeAll()
+      bookRegistryMock = nil
+      bookmarkCounter = 0
+    }
+
+    // MARK: - Test updateLocalBookmarks
+    
+    func testSyncServerBookmarksWithNoLocalBookmarks() throws {
```

Comment:
lemme know if I'm wrong, but I think this is testing the update of local bookmarks, not syncing with the server, correct? If so I'd rename this test to `testUpdateLocalBookmarksWithNoLocalBookmarks`, Same thing for the other tests below.